src/main/java/com/telefonica/iot/perseo/RulesServlet.java

```diff
@@ -160,26 +177,32 @@ protected void doDelete(HttpServletRequest request, HttpServletResponse response
         Utils.putCorrelatorAndTrans(request);
         response.setCharacterEncoding("UTF-8");
         response.setContentType("application/json;charset=UTF-8");
-        PrintWriter out = response.getWriter();
-        logger.info("delete rule " + request.getPathInfo());
-        //request.getPathInfo() returns null or the extra path information
-        //that follows the servlet path but precedes the query string and will
-        //start with a "/" character. So, we remove it with .substring(1)
-        String ruleName = request.getPathInfo();
-        if (ruleName == null) {
-            response.setStatus(400);
-            out.println("Deleting a rule require valid ruleName parameter");
-            out.close();
-            return;
+        
+        try {
+        	
+            logger.info(String.format("delete rule %s", request.getPathInfo()));
+            //request.getPathInfo() returns null or the extra path information
+            //that follows the servlet path but precedes the query string and will
+            //start with a "/" character. So, we remove it with .substring(1)
+            String ruleName = request.getPathInfo();
+            if (ruleName == null) {
+                response.setStatus(400);
+                response.getOutputStream().print("Deleting a rule require valid ruleName parameter");              
+                return;
+            }
+            Result r = RulesManager.delete(epService, ruleName.substring(1));
+
+            // Delete timed rule if necessary
+            TimeRulesStore.getInstance().removeTimeRule(ruleName.substring(1));
+
+            response.setStatus(r.getStatusCode());
+            response.getOutputStream().print(r.getMessage());           
+        }catch(Exception je) {        				
+			logger.error(String.format("error: %s" ,je));
```

Comment:
Bad indentation ?

Reply:
Fixed in d3cd3f4c88c67b3621b6714e4af99b3048ad605d